src/TriggerActivityMakerLowEnergyEvent.cpp

```diff
@@ -0,0 +1,269 @@
+          //for(auto tp : m_collection_window.inputs) dump_tp(tp);
+ 
+          // We have fulfilled our trigger condition, construct a TA and reset/flush the windows
+          // to ensure they're all in the same "time zone"!
+          output_ta.push_back(construct_ta(m_collection_window));
+          if (isZ) m_collection_window.reset(input_tp);
+          else m_collection_window.clear();
+          if (isU) m_induction1_window.reset(input_tp); 
+          else m_induction1_window.clear();
+          if (isY) m_induction2_window.reset(input_tp);   
+          else m_induction2_window.clear();
+  }
+
+  // Otherwise, slide the relevant window along using the current TP.
+  else {
+    if (isU) m_induction1_window.move(input_tp, m_window_length);
+    else if (isY) m_induction2_window.move(input_tp, m_window_length);
+    else if (isZ) m_collection_window.move(input_tp, m_window_length);
+  }
+  m_primitive_count++;
+
+  return;
+}
+
+void
+TriggerActivityMakerLowEnergyEvent::configure(const nlohmann::json& config)
+{
+  if (config.is_object()) {
+    if (config.contains("trigger_on_adc"))
+      m_trigger_on_adc = config["trigger_on_adc"];
+    if (config.contains("trigger_on_n_channels"))
+      m_trigger_on_n_channels = config["trigger_on_n_channels"];
+    if (config.contains("adc_threshold"))
+      m_adc_threshold = config["adc_threshold"];
+    if (config.contains("n_channels_threshold"))
+      m_n_channels_threshold = config["n_channels_threshold"];
+    if (config.contains("window_length"))
+      m_window_length = config["window_length"];
+    if (config.contains("trigger_on_adjacency"))
+      m_trigger_on_adjacency = config["trigger_on_adjacency"];
+    if (config.contains("adj_tolerance"))
+      m_adj_tolerance = config["adj_tolerance"];
+    if (config.contains("adjacency_threshold"))
+      m_adjacency_threshold = config["adjacency_threshold"];
+  }
+
+}
+
+TriggerActivity
+TriggerActivityMakerLowEnergyEvent::construct_ta(Window m_current_window) const
+{
+
+  TriggerPrimitive latest_tp_in_window = m_current_window.inputs.back();
+
+  TriggerActivity ta;
+  ta.time_start = m_current_window.time_start;
+  ta.time_end = latest_tp_in_window.time_start + latest_tp_in_window.time_over_threshold;
+  ta.time_peak = latest_tp_in_window.time_peak;
+  ta.time_activity = latest_tp_in_window.time_peak;
+  ta.channel_start = latest_tp_in_window.channel;
+  ta.channel_end = latest_tp_in_window.channel;
+  ta.channel_peak = latest_tp_in_window.channel;
+  ta.adc_integral = m_current_window.adc_integral;
+  ta.adc_peak = latest_tp_in_window.adc_peak;
+  ta.detid = latest_tp_in_window.detid;
+  ta.type = TriggerActivity::Type::kTPC;
+  ta.algorithm = TriggerActivity::Algorithm::kLowEnergyEvent;
+  ta.inputs = m_current_window.inputs;
+
+  return ta;
+}
+
+uint16_t
+TriggerActivityMakerLowEnergyEvent::check_adjacency(Window window) const
+{
+  // This function returns the adjacency value for the current window, where adjacency
```

Comment:
Comment blocks describing functionality should possibly be in Doxygen format, or in the .md file describing the algorithm

Reply:
The .md file describing the algorithm has been updated to outline the current development approach, and the basic logic of the TA and TC makers. I've also attempted a trimming of the adjacency checking logic in Doxygen format. If this is acceptable, I will go adjust the remainder of the code accordingly and commit at a later time.